server/controllers/checkAuth.ts

```diff
@@ -0,0 +1,12 @@
+import { Response } from 'express';
+
+const checkAuth = (req:any, res:Response) => {
+  res.status(200).json({
+    data: [{
+      id: req.user.id,
+      name: req.user.name,
+      isAdmin: req.user.isAdmin,
+    }],
+  });
+};
```

Comment:
You don't need to send the status code of 200 as it is the default and you already have the req.user object why you are creating a new one